### PR TITLE
enable sentry logging

### PR DESCRIPTION
**Pull request by zspitzer**
- renamed src .CFConfig files to .json5, no more complaining about comments
- providers now log to a custom log file, update-provider
- sentry logs out rest, application, update-provider and exception (would like to remove application as it's noisy)

### CODE REVIEW

File: compose.yaml [hunk 1]
```diff
@@ -30,6 +30,8 @@ services:
       - VIRTUAL_HOST=download.lucee.local
       - S3_DOWNLOAD_ACCESS_KEY_ID=${S3_DOWNLOAD_ACCESS_KEY_ID:-xxxxxx}
       - S3_DOWNLOAD_SECRET_KEY=${S3_DOWNLOAD_SECRET_KEY:-xxxxxx}
+      - SENTRY_ENV=${SENTRY_ENV:-xxxxxx}
+      - SENTRY_DSN=${SENTRY_DSN:-xxxxxx}
```
Comment (michaeloffner) [inline compose.yaml:34]: i would prefer no default values

File: compose.yaml [hunk 2]
```diff
@@ -47,4 +49,6 @@ services:
       - S3_EXTENSION_SECRET_KEY=${S3_EXTENSION_SECRET_KEY:-xxxxxx}
       - S3_EXTENSION_ACCESS_KEY_ID=${S3_EXTENSION_ACCESS_KEY_ID:-xxxxxx}
       - S3_DOWNLOAD_SECRET_KEY=${S3_DOWNLOAD_SECRET_KEY:-xxxxxx}
-      - S3_DOWNLOAD_ACCESS_KEY_ID=${S3_DOWNLOAD_ACCESS_KEY_ID:-xxxxxx}
\ No newline at end of file
+      - S3_DOWNLOAD_ACCESS_KEY_ID=${S3_DOWNLOAD_ACCESS_KEY_ID:-xxxxxx}
+      - SENTRY_ENV=${SENTRY_ENV:-xxxxxx}
+      - SENTRY_DSN=${SENTRY_DSN:-xxxxxx}
```
Comment (michaeloffner) [inline compose.yaml:54]: no default

File: devops/.CFconfig-download.json5 [hunk 3]
```diff
@@ -131,17 +131,38 @@
             "level": "error",
             "layout": "classic"
         },
+        "update-provider": {
+            "appenderArguments": "environment:%7Benv%2ESENTRY%5FENV%7D;dist:;extras:;dsn:%7Benv%2ESENTRY%5FDSN%7D;debug:true;tags",
+            "level": "error",
+            "appenderClass": "org.lucee.extension.sentry.log.log4j.SentryAppenderLog4j2",
+            "appenderBundleName": "sentry.extension",
+            "appenderBundleVersion": "5.5.2.12-BETA",
```
Comment (michaeloffner) [inline devops/.CFconfig-download.json5:139]: i added a newer version [5.5.2.15 (Dec 6, 2023)](https://ext.lucee.org/sentry-extension-5.5.2.15.lex)

File: devops/.CFconfig-download.json5 [hunk 4]
```diff
@@ -131,17 +131,38 @@
...
+        },
         "exception": {
-            "appender": "resource",
-            "appenderArguments": "path:{lucee-config}/logs/exception.log",
+            "appenderArguments": "environment:%7Benv%2ESENTRY%5FENV%7D;dist:;extras:;dsn:%7Benv%2ESENTRY%5FDSN%7D;debug:true;tags",
             "level": "error",
-            "layout": "classic"
+            "appenderClass": "org.lucee.extension.sentry.log.log4j.SentryAppenderLog4j2",
+            "appenderBundleName": "sentry.extension",
+            "appenderBundleVersion": "5.5.2.12-BETA",
```
Comment (michaeloffner) [inline devops/.CFconfig-download.json5:148]: [5.5.2.15 (Dec 6, 2023)](https://ext.lucee.org/sentry-extension-5.5.2.15.lex)

File: devops/.CFconfig-download.json5 [hunk 5]
```diff
@@ -131,17 +131,38 @@
...
+            "layoutClass": "lucee.commons.io.log.log4j2.layout.ClassicLayout",
+            "layoutArguments": ""
         },
         "application": {
+            "appenderArguments": "environment:%7Benv%2ESENTRY%5FENV%7D;dist:;extras:;dsn:%7Benv%2ESENTRY%5FDSN%7D;debug:true;tags",
+            "level": "info",
+            "appenderClass": "org.lucee.extension.sentry.log.log4j.SentryAppenderLog4j2",
+            "appenderBundleName": "sentry.extension",
+            "appenderBundleVersion": "5.5.2.12-BETA",
```
Comment (michaeloffner) [inline devops/.CFconfig-download.json5:157]: [5.5.2.15 (Dec 6, 2023)](https://ext.lucee.org/sentry-extension-5.5.2.15.lex)

File: devops/.CFconfig-download.json5 [hunk 6]
```diff
@@ -267,12 +301,12 @@
             "id": "8D7FB0DF-08BB-1589-FE3975678F07DB17",
             "name": "Compress Tags",
             "version": "1.0.0.15"
-        },
+        } /*
         {
-            "id": "83062C18-FA1F-4647-815BB663BCF98AC0",
-            "name": "Sentry",
-            "version": "5.5.2.12-BETA"
-        }
```
Comment (michaeloffner) [inline devops/.CFconfig-download.json5:275]: why removing it?

File: devops/.CFconfig-update.json5 [hunk 7]
```diff
@@ -131,12 +131,17 @@
             "level": "error",
             "layout": "classic"
         },
+        "update-provider": {
+            "appenderArguments": "environment:%7Benv%2ESENTRY%5FENV%7D;dist:;extras:;dsn:%7Benv%2ESENTRY%5FDSN%7D;debug:true;tags",
+            "level": "error",
+            "appenderClass": "org.lucee.extension.sentry.log.log4j.SentryAppenderLog4j2",
+            "appenderBundleName": "sentry.extension",
+            "appenderBundleVersion": "5.5.2.12-BETA",
```
Comment (michaeloffner) [inline devops/.CFconfig-update.json5:139]: [5.5.2.15 (Dec 6, 2023)](https://ext.lucee.org/sentry-extension-5.5.2.15.lex)

File: devops/.CFconfig-update.json5 [hunk 8]
```diff
@@ -212,10 +207,13 @@
...
         },
         "rest": {
-            "appender": "resource",
-            "appenderArguments": "path:{lucee-config}/logs/rest.log",
+            "appenderArguments": "environment:%7Benv%2ESENTRY%5FENV%7D;dist:;extras:;dsn:%7Benv%2ESENTRY%5FDSN%7D;debug:true;tags",
             "level": "error",
-            "layout": "classic"
+            "appenderClass": "org.lucee.extension.sentry.log.log4j.SentryAppenderLog4j2",
+            "appenderBundleName": "sentry.extension",
+            "appenderBundleVersion": "5.5.2.12-BETA",
```
Comment (michaeloffner) [inline devops/.CFconfig-update.json5:214]: [5.5.2.15 (Dec 6, 2023)](https://ext.lucee.org/sentry-extension-5.5.2.15.lex)

File: update/UpdateProvider.cfc [hunk 9]
```diff
@@ -23,6 +24,11 @@
 		directoryCreate(variables.artDirectory);
 	variables.extDirectory="/var/www/extension/extension/"; // TODO make more dynamic
 
+	private function logger( string text, string stack="", type="info" ){
+		var log = arguments.text & chr(13) & chr(10) & callstackGet('string') & chr(13) & chr(10) & arguments.stack;
+		WriteLog( text=log, type=arguments.type, log=variables.providerLog );
```
Comment (michaeloffner) [inline update/UpdateProvider.cfc:29]: you can also feed an exception into log with the attribute "exception"

File: update/UpdateProvider.cfc [hunk 10]
```diff
@@ -522,8 +527,11 @@ try{
 	        content variable="#bin#" type="application/zip"; // in future version this should be handled with producer attribute
 		}
 }
-catch(e) { return e;}
-	}
+catch(e) { 
+	logger (text=e.message, stack=e.stacktrace, type="error");
```
Comment (michaeloffner) [inline update/UpdateProvider.cfc:531]: better pass the exception itself,easier and better result

File: update/UpdateProvider.cfc (hunk 9 above)
Comment (zspitzer) [reply update/UpdateProvider.cfc:29]: ah yeah, does the sentry extension do something sexy with that?

**Pull request merged**